pages/ideaDetails/[id].js

```diff
@@ -0,0 +1,95 @@
+import Header from "@/components/Header";
+import Image from "next/image";
+import Link from "next/link";
+import styled from "styled-components";
+
+import { v4 as uuidv4 } from "uuid";
+import { ideas as defaultIdeas } from "@/lib/db";
+import { useState } from "react";
+import { useRouter } from "next/router.js";
+
+const StyledArticle = styled.article`
+  display: flex;
+  flex-wrap: wrap;
+  gap: 1rem;
+  list-style: none;
+  padding: 0;
+  margin-top: 1rem;
+  flex-direction: column;
+`;
+
+const StyledImage = styled(Image)`
+  border-radius: 0.5rem;
+`;
+
+const StyledContainer = styled.div`
+  display: flex;
+  gap: 1rem;
+`;
+
+const Instruction = styled.ol`
+  padding-left: 1rem;
+`;
+
+const Hashtags = styled.ul`
+  list-style: none;
+  display: flex;
+  gap: 0.5rem;
+  flex-wrap: wrap;
+`;
+
+const Items = styled.ul`
+  list-style: none;
+  border: 1px solid black;
+  border-radius: 0.5rem;
+  padding: 0.5rem;
+`;
+
+export default function IdeaDetails() {
+  const [ideas, setIdeas] = useState(defaultIdeas);
+  const router = useRouter();
+
+  const { id } = router.query;
+
+  const ideaDetails = ideas.find((idea) => idea.id === id);
+
+  if (!ideaDetails) {
+    return <h2>Loading...</h2>;
+  }
+
+  const { instructions, items, hashtags } = ideaDetails;
+
+  return (
+    <>
+      <Header />
+      <StyledArticle>
+        <h2>{ideaDetails.title}</h2>
+        <StyledContainer>
+          <StyledImage
+            src={ideaDetails.image}
+            alt={ideaDetails.title}
+            width={150}
+            height={120}
+          />
+          <Items>
+            {items.map((item) => (
+              <li key={uuidv4()}>{item}</li>
```

Comment:
Like this, the function will be called every time the component renders. This is not really amazing performance-wise, but it's fine for now. It's usually better to have static identifiers, unless the component is trivial and doesn't usually need to be re-rendered, e.g. is static itself.

Reply:
We understand your suggestion, but we need your input on how to create a static identifier out of the uuid. We could make it static by turning them into objects, but we would want to know what you would suggest?